WcaOnRails/spec/controllers/registrations_controller_spec.rb

```diff
@@ -161,6 +161,15 @@
       patch :update, id: registration.id, registration: { status: :accepted }
       expect(registration.reload.accepted?).to eq true
     end
+
+    it "cannot accept own registration for a competition that isn't visible" do
+      competition.showAtAll = 0
+      competition.save!
```

Comment:
You're probably right...I don't know for sure when to raise exceptions or not :P